lib/withReactiveQuery.js

```diff
@@ -14,24 +16,48 @@ export default function withReactiveContainer(handler, config, QueryComponent) {
     return withTracker((props) => {
         const query = handler(props);
 
-        const subscriptionHandle = query.subscribe({
-            onStop(err) {
-                if (err) {
-                    subscriptionError.set(err);
-                }
-            },
-            onReady() {
-                subscriptionError.set(null);
-            }
-        });
+        let isLoading
+        let data
+        let error
 
-        const isReady = subscriptionHandle.ready();
+        // For server-side-rendering, immediately fetch the data
+        // and save it in the data store for this request
+        
+        if(Meteor.isServer) {
+            data = query.fetch();
+            isLoading = false;
+            props.dataStore.add(query, data);
+        } else {
+            const subscriptionHandle = query.subscribe({
+                onStop(err) {
+                    if (err) {
+                        subscriptionError.set(err);
+                    }
+                },
+                onReady() {
+                    subscriptionError.set(null);
+                }
+            });
+    
+            isLoading = !subscriptionHandle.ready();
 
-        const data = query.fetch();
+            // Check the SSR query store for data 
+            if(Meteor.isClient && window && window.grapherQueryStore) {
+                const ssrData = DataHydrator.getQueryData(query);
```

Comment:
I now understand why you are doing the self-destruct thingie. But a healthier approach would be to set a flag once first hydration has been made, and do the check in here. I don't know something like `window.grapher.hydrationCompleted = true`  

And pls use consistent styles `if[SPACE](...)`